lib/datadog/statsd/sender.rb

```diff
@@ -23,20 +30,28 @@ def rendez_vous
         queue = (Thread.current[:statsd_sync_queue] ||= Queue.new)
         # tell sender-thread to notify us in the current
         # thread's queue
-        message_queue.push(queue)
+        @message_queue.push(queue)
         # wait for the sender thread to send a message
         # once the flush is done
         queue.pop
       end
 
       def add(message)
-        raise ArgumentError, 'Start sender first' unless message_queue
+        raise ArgumentError, 'Start sender first' unless @message_queue
 
-        message_queue << message
+        # if the thread does not exist, we are probably running in a forked process,
+        # empty the message queue (these messages belong to parent process) and spawn
+        # a new companion thread.
+        if !@sender_thread.alive?
+          @message_queue = nil
+          start
+        end
```

Comment:
*  (1) (3) 👍 
* About (2), we would naively try to re-create a thread. I could maybe track how many times we are creating a new thread per second, and emit a warning if it is quite high, but not sure if that's worth. Except from emitting a log message, I don't think there is much we can do since it could be normal in some environments with a lot of forks.

Reply:
Yeah, I think a log message would be nice. The throttling mechanism is an interesting idea, but perhaps not worth the complexity without a clearer use case than "well I what if the thread dies for unknown reasons".